gax/src/main/java/com/google/api/gax/rpc/mtls/MtlsUtils.java

```diff
@@ -0,0 +1,178 @@
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.google.api.gax.rpc.mtls;
+
+import com.google.api.client.json.JsonParser;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.SecurityUtils;
+import com.google.api.core.BetaApi;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.util.List;
+
+/** Utilities for mutual TLS. */
+@BetaApi
+public class MtlsUtils {
+  @VisibleForTesting
+  static class DefaultMtlsProvider implements MtlsProvider {
+    private static final String DEFAULT_CONTEXT_AWARE_METADATA_PATH =
+        System.getProperty("user.home") + "/.secureConnect/context_aware_metadata.json";
+
+    /** GOOGLE_API_USE_CLIENT_CERTIFICATE environment variable. */
+    public static final String GOOGLE_API_USE_CLIENT_CERTIFICATE =
+        "GOOGLE_API_USE_CLIENT_CERTIFICATE";
+
+    /** GOOGLE_API_USE_MTLS_ENDPOINT environment variable. */
+    public static final String GOOGLE_API_USE_MTLS_ENDPOINT = "GOOGLE_API_USE_MTLS_ENDPOINT";
+
+    interface EnvironmentProvider {
+      String getenv(String name);
+    }
+
+    static class SystemEnvironmentProvider implements EnvironmentProvider {
+      @Override
+      public String getenv(String name) {
+        return System.getenv(name);
+      }
+    }
+
+    DefaultMtlsProvider() {
+      this(new SystemEnvironmentProvider(), DEFAULT_CONTEXT_AWARE_METADATA_PATH);
+    }
+
+    private EnvironmentProvider envProvider;
+    private String metadataPath;
+
+    /** Cache the key store so we don't create a new one every time getKeyStore is called. */
+    private KeyStore keyStore = null;
+
+    @VisibleForTesting
+    DefaultMtlsProvider(EnvironmentProvider envProvider, String metadataPath) {
+      this.envProvider = envProvider;
+      this.metadataPath = metadataPath;
+    }
+
+    @Override
+    public boolean useMtlsClientCertificate() {
+      String useClientCertificate = envProvider.getenv(GOOGLE_API_USE_CLIENT_CERTIFICATE);
+      return "true".equals(useClientCertificate);
```

Comment:
Should we also support other variants of this, like "True", "TRUE", "yes", "1"?

Reply:
Alternatively, `useClientCertificate.toLower()`.

Reply:
No, we only support the lower case "true".